numpy/lib/npyio.py

```diff
@@ -791,7 +792,14 @@ def loadtxt(fname, dtype=float, comments='#', delimiter=None,
     """
     # Type conversions for Py3 convenience
     if comments is not None:
-        comments = asbytes(comments)
+        if isinstance(comments, (np.compat.basestring, np.compat.bytes)):
```

Comment:
I simply meant that I am not well versed in numpy's compatibility layer, so I'd like someone else on the team to have a look.